workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/internal/flows/distribution/DistributionUtilities.kt

```diff
@@ -62,21 +62,21 @@ fun ServiceHub.addPartyToDistributionList(party: Party, linearId: UniqueIdentifi
 }
 
 @Suspendable
-fun FlowLogic<*>.addToDistributionList(tokens: List<AbstractToken<TokenPointer<*>>>) {
-    tokens.forEach { token ->
-        val tokenType = token.tokenType
+fun FlowLogic<*>.addToDistributionList(tokens: List<AbstractToken>) {
+    tokens.filter { it.tokenType as? TokenPointer<*> != null }.forEach { token ->
```

Comment:
You can use filterIsInstance, then you don't need to do a cast later

Reply:
Achually, you can't because we are not filtering on `it` but rather `it.tokenType` so Kotlin isn't smart enough to remember what `it.tokenType` is.